dk-installer.py

```diff
@@ -496,21 +512,29 @@ def start_cmd(self, *cmd, raise_on_non_zero=True, env=None, **popen_args):
         started = time.time()
         self._cmd_idx += 1
 
-        LOG.debug("Command [%04d]: [%s]", self._cmd_idx, " ".join(cmd))
+        flattened_cmd = [str(item) for sublist in cmd for item in
+                         (sublist if isinstance(sublist, tuple) else [sublist])]
+
+        LOG.debug("Command [%04d]: [%s]", self._cmd_idx, " ".join(flattened_cmd))
```

Comment:
I'm not a fan of this logic, especially inside this function that is already doing too much things.

Instead, you can use the star operator (`*`) to expand the list of command pieces inside another list of a function call. That would be much cleaner.

Also, if you go with the approach I'm suggesting later of creating a function to run the docker commands, most of the call sites that are requiring this will vanish.